rbhl/api.py

```diff
@@ -144,20 +144,39 @@ def trial_data(self, episode, trial_num, peak_flow_days):
             "treatments": self.get_treatments(days),
             "overrall_mean": self.get_overrall_mean(peak_flow_days),
             "pef_mean": pef,
-            "notes": self.get_notes(peak_flow_days)
+            "notes": self.get_notes(peak_flow_days),
         }
 
+    def get_order(self, by_trial):
+        """
+        In some of the import data trial number is not
+        correlated to trial start date.
+
+        This will return a list of trial numbers as ordered by start date.
+        """
+        trial_num_and_min_date = []
+        for trial_num, trial_data in by_trial.items():
+            trial_num_and_min_date.append({
+                "trial_num": trial_num,
+                "start_date": min([i["date"] for i in trial_data["days"]])
+            })
+        sorted_by_date = sorted(trial_num_and_min_date, key=lambda x: x["start_date"])
+        return [i["trial_num"] for i in sorted_by_date]
+
     @episode_from_pk
     def retrieve(self, request, episode):
         groups = itertools.groupby(
             episode.peakflowday_set.order_by("trial_num", "day_num"),
             key=lambda x: x.trial_num
         )
-        result = {
+        by_trial = {
             trial_num: self.trial_data(episode, trial_num, list(group))
             for trial_num, group in groups
         }
-        return json_response(result)
+        return json_response({
```

Comment:
I've had a look, I'm not sure I understand why this can't be a list of trial data - why do we need the object at all?

Reply:
@davidmiller changed so that it is returning a list, its better this way but a much bigger change.